backend/tests/unit/protocols/backend.txt

```diff
@@ -0,0 +1,106 @@
+Scenario: The backend receiver's input event correctly reports whether it has data
+Given: A ReceiveEvent object constructed with default parameters
+When: the object is not modified
+Then: has_data function returns false   
+
+Scenario: The backend receiver's input event correctly reports whether it has data
+Given:  A ReceiveEvent object constructed with a value for the time parameter
+When: a new value is set for the time field of the object
+Then: has_data function returns true
+
+Scenario: The backend receiver's input event correctly reports whether it has data
+Given: A ReceiveEvent object constructed with values for time and mcu parameter
+When: a new value is set for the mcu field of the object
+Then: has_data function returns true
+
+Scenario: The backend receiver's input event correctly reports whether it has data
+Given: A ReceiveEvent object constructed with values for time, mcu and frontend parameter
+When: a new value is set for the frontend field of the object
+Then: has_data function returns true
+
+Scenario: The backend's output event correctly reports whether it has data
+Given: A OutputEvent object constructed with default parameters
+When: the object is not modified
+Then: has_data function returns false
+
+Scenario: The backend's output event correctly reports whether it has data
+Given: A OutputEvent object constructed with a value for the mcu parameter
+When: a new value is set for the mcu field of the object
+Then: has_data function returns true
+
+Scenario: The backend's output event correctly reports whether it has data
+Given: A OutputEvent object constructed with a value for the mcu and frontend parameter
+When: a new value is set for the frontend field of the object
+Then: has_data function returns true
+
+Scenario: The backend's output event correctly reports whether it has data
+Given: A OutputEvent object constructed with a value for the mcu, frontend and file parameter
+When: a new value is set for the file field of the object
+Then: has_data function returns true
+
+Scenario: The input event sent by backend correctly reports whether it has data
+Given: a announcement object constructed with default parameters
+When: the object is not modified
+Then: has_data function returns false
+
+Scenario: The input event sent by backend correctly reports whether it has data
+Given: a announcement object constructed a value for the message field
+When: a new value is set for the message field of the object
+Then: has_data function returns true
+
+#ReceiveFilter
+
+Scenario: The receive filter's output correctly reports whether it has data
+Given: a ReceiveFilter object which has not yet received any events
+When: a receive event with no data is passed as input to the receive filter
+Then: the output of the receive filter is None
+
+Scenario: The receive filter's output correctly reports whether it has data
+Given: A ReceiveEvent object constructed with values for the time and mcu parameter
```

Comment:
On these tests, we need to be given a receive filter object. If you are constructing a different ReceiveEvent object for each test, it might make more sense to define it in the When. So, for example:

Given: A ReceiveFilter object constructed with ___ and which has an empty internal input events buffer
(Then: The output method of the ReceiveFilter object returns None each time it is called)
When: A ReceiveEvent object with values for ___ is given as input to the ReceiveFilter object
Then: The output method of the ReceiveFilter object returns None each time it is called